arangod/MMFiles/MMFilesWalRecoverState.cpp

```diff
@@ -911,12 +911,35 @@ bool MMFilesWalRecoverState::ReplayMarker(MMFilesMarker const* marker,
         }
 
         // turn off sync temporarily if the database or collection are going to
-        // be
-        // dropped later
+        // be dropped later
         bool const forceSync = state->willViewBeDropped(databaseId, viewId);
+        
+        VPackSlice nameSlice = payloadSlice.get("name");
+        if (nameSlice.isString() && !nameSlice.isEqualString(view->name())) {
+          std::string name = nameSlice.copyString();
+          // check if other view exists with target name
+          std::shared_ptr<arangodb::LogicalView> other = vocbase->lookupView(name);
+          if (other != nullptr) {
+            if (other->id() == view->id()) {
+              LOG_TOPIC(TRACE, arangodb::Logger::ENGINES)
+              << "view " << viewId << " in database " << databaseId
+              << " was already renamed; moving on";
+              break;
+            }
+            vocbase->dropView(other->id(), true);
```

Comment:
should the code really drop the view with the same name instead of returning duplicate-name-error?

Reply:
that was what the original rename marker code did. I don't know if that is the best way, but collections seem to  have the same conflict resolution logic

Reply:
quite dangerous if user-triggerable